File: LintCode/26. Remove Duplicates from Sorted Array.cpp
// https://leetcode.com/problems/remove-duplicates-from-sorted-array/
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int n = nums.size();
        if(n == 0){
            return 0;
        }
        
        int j = 0;
        for(int i = 1; i < n; ++i)
        {
            if(nums[i] != nums[j])
            {
                nums[++j] = nums[i];
            }
        }
        return j+1;
    }
};

File: LeetCode/Python3/26.RemoveDuplicatesfromSortedArray.py
# https://leetcode.com/problems/remove-duplicates-from-sorted-array/

# Time complexity:O(n)
# Space complexity:O(1)

class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums:
            return 0
        j = 0  # recorder zero
        for i in range(len(nums)):
            if nums[j] != nums[i]:
                j += 1
                nums[j] = nums[i]
        return j+1
